tools/ci/maven-utils.sh

```diff
@@ -73,7 +73,7 @@ function collect_coredumps {
 	echo "Searching for .dump, .dumpstream and related files in '$SEARCHDIR'"
 	for file in `find $SEARCHDIR -type f -regextype posix-extended -iregex '.*\.hprof|.*\.dump|.*\.dumpstream|.*hs.*\.log|.*/core(.[0-9]+)?$'`; do
 		echo "Moving '$file' to target directory ('$TARGET_DIR')"
-		mv $file $TARGET_DIR/
+		mv $file $TARGET_DIR/$(echo $file | tr "/" "-")
```

Comment:
shouldn't be possible to have forward slashes in file names on unix

Reply:
The problem I'm solving here is the following: Before this change, the YARN test profile has the following output:
```
Moving '/__w/2/s/flink-python/target/surefire-reports/2020-05-25T09-59-37_808.dumpstream' to target directory ('/__w/2/s/tools/artifacts')
Moving '/__w/2/s/flink-filesystems/flink-azure-fs-hadoop/target/surefire-reports/2020-05-25T09-59-37_808.dumpstream' to target directory ('/__w/2/s/tools/artifacts')
Moving '/__w/2/s/flink-examples/flink-examples-streaming/target/surefire-reports/2020-05-25T09-59-37_808.dumpstream' to target directory ('/__w/2/s/tools/artifacts')
```

Since all the files have the same name, we will only provide the last file to the user, not all of them.
With this change, we are effectively renaming the file to contain the path, to make the filenames unique.

Reply:
how could there be dumpstreams in multiple modules? Aren't we failing on the first test failure?

Reply:
I just checked, and these dumpstream files just contain for example the following
```
# Created at 2020-05-25T10:24:38.843
Picked up JAVA_TOOL_OPTIONS: -XX:+HeapDumpOnOutOfMemoryError
```

I guess the surefire plugin is creating a dumpstream file if it receives any output, but it only fails if it receives unexpected output?

Reply:
hmm, not sure. Maybe this happens because the JVM writes that to the STDERR. That's annoying :/

This is probably fine then; I was initially worried we might get so long filenames that you wouldn't be able to extract the logs on Windows.